### DISCUSSION TITLE
How are KShape cluster centers calculated? I thought they would be the arithmic mean the data points per time step, but it seems not

### DISCUSSION THREAD

**Discussion by sjowtajm** [tslearn-team/tslearn#506] [Q&A]
Hey

After implementing KMeans cluster I am now looking at KShape to compare the results between the two. I am using both through Anaconda.Navigator and I have tslearn 0.6.3 installed.

While I can see that the cluster centers from KMeans are the mean in each group, I don't understand the result from KShape.

The input data I have are daily time series with hourly resolution, i.e. 24 data points per time series.

Look at the documentation for [KShape ](https://tslearn.readthedocs.io/en/stable/gen_modules/clustering/tslearn.clustering.KShape.html) it says that the cluster centers are the centroids, which I interpret as the center = average value of the time series?

But when I try a few time series as input, the centers look wierd. See some figures below:

**With 4 times series it seems to pick one of the time series in cluster 1 as the centroid**

![image](https://github.com/tslearn-team/tslearn/assets/55394059/32dd5566-41ff-4b11-9bd7-2fbc60aa38a0)

**With 10 time series the centroid seems to be shifted left one step for all clusters**

![image](https://github.com/tslearn-team/tslearn/assets/55394059/d1f16cfa-8eb8-4ead-9b02-b09798ce8891)

**With 40 time series it seems to be even more shifted? The sole time series in cluster 4 is shifted three time steps wrong to the left**

![image](https://github.com/tslearn-team/tslearn/assets/55394059/346ab812-d854-4ebc-bc6c-1b9a55e0e1c7)

These three examples are on z-normalized data, which seems to be what you must input. It does not say that in the documents, but if I input non-normalized data the centroid become even more chaotic:

![image](https://github.com/tslearn-team/tslearn/assets/55394059/52e3758c-c282-4607-b4c9-fbf89ff044b8)

---

Is this a bug, or have I done something wrong? The data I input is a DataFrame called "cluster_data" with 24 columns and where each row is a separate time series, then I call:

```
 cluster_data = TimeSeriesScalerMeanVariance().fit_transform(cluster_data) # scales data to have mean = 0, std = 1 with standard settings (z-normalization)

        cluster_data = pd.DataFrame(cluster_data.squeeze())

        km = KShape(n_clusters=number_of_clusters, verbose=True, random_state=0)

        y_pred = km.fit_predict(cluster_data)

        cluster_centers = pd.DataFrame(km.cluster_centers_[:,:,0])
```
